File: backend/utils/format_download_speed.py
def format_download_speed(bytes_per_sec: float | int):
  kb = bytes_per_sec / 1024
  mb = bytes_per_sec / (1024 * 1024)

  if mb >= 1:
    return f"{mb:.1f} MB/s"

  return f"{kb:.1f} kB/s"

File: frontend/features/view-downloads/utils/formatDownloadSpeed.ts
/**
 * Formats a download speed given in bytes per second to a human-readable string.
 *
 * @param bytesPerSec The download speed in bytes per second.
 * @returns The formatted string.
 */
export default function formatDownloadSpeed(
  bytesPerSec: number | null
): string {
  if (bytesPerSec === null) return "";

  const kb = bytesPerSec / 1024;
  const mb = bytesPerSec / (1024 * 1024);

  if (mb >= 1) {
    return `${mb.toFixed(1)} MB/s`;
  }

  return `${kb.toFixed(1)} kB/s`;
}
